searx/plugins/__init__.py

```diff
@@ -77,6 +87,81 @@ def call(self, ordered_plugin_list, plugin_type, request, *args, **kwargs):
         return ret
 
 
+def load_external_plugins(plugin_names):
+    plugins = []
+    for name in plugin_names:
+        logger.debug('loading plugin: {0}'.format(name))
+        try:
+            pkg = import_module(name)
+        except Exception as e:
+            logger.critical('failed to load plugin module {0}: {1}'.format(name, e))
+            exit(3)
+
+        pkg.__base_path = dirname(abspath(pkg.__file__))
+
+        fix_package_resources(pkg, name)
+
+        plugins.append(pkg)
+        logger.debug('plugin "{0}" loaded'.format(name))
+    return plugins
+
+
+def check_resource(base_path, resource_path, name, target_dir, plugin_dir):
+        dep_path = join(base_path, resource_path)
+        file_name = basename(dep_path)
+        resource_path = join(target_dir, file_name)
+        if not exists(resource_path) or sha_sum(dep_path) != sha_sum(resource_path):
+            try:
+                copyfile(dep_path, resource_path)
```

Comment:
I guess its better to explicit  catch [OSError](https://docs.python.org/3/library/exceptions.html#OSError) / compare [Exception hierarchy](https://docs.python.org/3/library/exceptions.html#exception-hierarchy).

Its the same with the other OS related try/except blocks.

Reply:
I would copy atime_ns and mtime_ns, so the weak ETags (generated by the HTTP server) do not change.

Reply:
> I guess its better to explicit catch OSError / compare Exception hierarchy.

Why?

Reply:
Explicit catch what you can handle / never catch exceptions you do not handle.